exonum-java-binding/time-oracle/src/main/java/com/exonum/binding/time/TimeSchemaProxy.java

```diff
@@ -16,74 +16,74 @@
 
 package com.exonum.binding.time;
 
-import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Preconditions.checkArgument;
 
 import com.exonum.binding.common.crypto.PublicKey;
 import com.exonum.binding.common.serialization.Serializer;
 import com.exonum.binding.common.serialization.StandardSerializers;
 import com.exonum.binding.core.storage.database.View;
 import com.exonum.binding.core.storage.indices.EntryIndexProxy;
+import com.exonum.binding.core.storage.indices.MapIndex;
 import com.exonum.binding.core.storage.indices.ProofMapIndexProxy;
-import com.exonum.binding.core.util.LibraryLoader;
+import com.exonum.binding.messages.Runtime.ArtifactId;
+import com.exonum.binding.messages.Runtime.InstanceSpec;
 import java.time.ZonedDateTime;
 
 class TimeSchemaProxy implements TimeSchema {
 
-  static {
-    LibraryLoader.load();
-  }
+  private static final int RUST_RUNTIME_ID = 0;
+  private static final String EXONUM_TIME_ARTIFACT_NAME_PREFIX = "exonum-time:";
 
   private static final Serializer<PublicKey> PUBLIC_KEY_SERIALIZER =
       StandardSerializers.publicKey();
   private static final Serializer<ZonedDateTime> ZONED_DATE_TIME_SERIALIZER =
       UtcZonedDateTimeSerializer.INSTANCE;
 
-  private final View dbView;
+  private final View view;
+  private final String name;
 
-  TimeSchemaProxy(View dbView) {
+  TimeSchemaProxy(View view, String name) {
+    this.name = name;
+    this.view = view;
     checkIfEnabled();
-    this.dbView = dbView;
+  }
+
+  private void checkIfEnabled() {
+    MapIndex<String, InstanceSpec> serviceInstances = new RuntimeSchema(view).serviceInstances();
+    checkArgument(serviceInstances.containsKey(name), "No service instance "
+        + "with the given name (%s) started.");
```

Comment:
`    checkArgument(serviceInstances.containsKey(name), "No service instance "
        + "with the given name (%s) started.", name);`